src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/FeatureFlag/FeatureFlagCollectionType.php

```diff
@@ -0,0 +1,120 @@
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Configure\AdvancedParameters\FeatureFlag;
+
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use PrestaShopBundle\Form\FormCloner;
+use Symfony\Component\Form\Event\PreSetDataEvent;
+use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Represents the form used to manage feature flags state.
+ * There is one SwitchType per existing feature flag.
+ */
+class FeatureFlagCollectionType extends TranslatorAwareType
+{
+    /**
+     * @var bool
+     */
+    protected $isMultiShopUsed;
+
+    /**
+     * @var FormCloner
+     */
+    protected $formCloner;
+
+    /**
+     * FeatureFlagCollectionType constructor.
+     *
+     * @param TranslatorInterface $translator
+     * @param array $locales
+     * @param bool $isMultiShopUsed
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        bool $isMultiShopUsed,
+        FormCloner $formCloner
+    ) {
+        parent::__construct($translator, $locales);
+        $this->isMultiShopUsed = $isMultiShopUsed;
+        $this->formCloner = $formCloner;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('feature_flags', CollectionType::class, [
+                'entry_type' => FeatureFlagType::class,
+                'label' => false,
+                'required' => false,
+                'allow_extra_fields' => true,
+                'attr' => [
+                    'data-is-multi-shop-used' => $this->isMultiShopUsed ? '1' : '0',
+                ],
+            ])
+            ->addEventListener(FormEvents::PRE_SET_DATA, [$this, 'addSubmitButton']);
```

Comment:
I needed data from the form data provider to configure the submit button with the right data attributes...

And it seems like this data is only accessible from this event...  I'm not sure if there was an easier way ? 🤔